### PR TITLE
Fixes #13 wrap unlink with ErrorHandler for gc on concurrent requests

### LINKED ISSUE #13: Silence PHP Warning: unlink() in GC
Sometimes I get: _PHP Warning: unlink()_ due to concurrent requests on server. Thus `unlink($file->getPathname());` in method _\Zend\Captcha\Image::gc()_ should be silenced with: `@unlink($file->getPathname());`.

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes #13

### CODE REVIEW

File: src/Image.php [hunk 1]
```diff
@@ -609,7 +609,7 @@ protected function gc()
                 if (file_exists($file->getPathname()) && $file->getMTime() < $expire) {
                     // only deletes files ending with $this->suffix
                     if (substr($file->getFilename(), -($suffixLength)) == $this->suffix) {
-                        unlink($file->getPathname());
+                        @unlink($file->getPathname());
```
Comment (weierophinney) [inline src/Image.php]: Wrap this in an error handler, please; error suppression is expensive, and makes testing more difficult.

Comment (samsonasik) [reply src/Image.php]: wrap with `Zend\Stdlib\ErrorHandler` ?

**Comment by samsonasik** [comment]
@weierophinney I've wrapped the `unlink()` with `Zend\Stdlib\ErrorHandler::start()` and `Zend\Stdlib\ErrorHandler::stop()`, please let me know if any something I missed. 

Thank you.

**Comment by akrabat** [comment]
@samsonasik Can you open this PR against develop please to ensure that travis passes?

**Comment by samsonasik** [comment]
@akrabat I rebased against develop and change target repo to develop branch

**Comment by samsonasik** [comment]
@akrabat travis green \m/

File: src/Image.php [hunk 2]
```diff
@@ -609,7 +609,9 @@ protected function gc()
                 if (file_exists($file->getPathname()) && $file->getMTime() < $expire) {
                     // only deletes files ending with $this->suffix
                     if (substr($file->getFilename(), -($suffixLength)) == $this->suffix) {
+                        ErrorHandler::start();
                         unlink($file->getPathname());
+                        ErrorHandler::stop();
```
Comment (akrabat) [inline src/Image.php:614]: Pretty sure that this needs to be `ErrorHandler::stop(true);` or you need to check the return value from `stop()` and do something with it. Thoughts @weierophinney?

Comment (evpav) [reply src/Image.php:614]: @akrabat, isn't this intentional, to just silence the warning?

Comment (weierophinney) [reply src/Image.php:614]: @akrabat `ErrorHandler::stop()` takes an optional boolean, `false` by default, to indicate whether or not it should _throw an exception_ if an error was caught. In this particular case, we're willing to ignore any warnings raised by `unlink()`, so no argument is necessary.

**Comment by weierophinney** [comment]
Thanks, @samsonasik!

**Pull request merged**